django/core/mail/message.py

```diff
@@ -354,7 +354,9 @@ def _create_attachments(self, msg):
             encoding = self.encoding or settings.DEFAULT_CHARSET
             body_msg = msg
             msg = SafeMIMEMultipart(_subtype=self.mixed_subtype, encoding=encoding)
-            if self.body:
+            # Attach the body_msg if self.body isn't empty or
+            # body_msg is a list of SafeMIMEText or SafeMIMEMultipart objects.
```

Comment:
Now I see that you originally copied `EmailMessage` from the Python docs for `is_multipart()`. I had confused that with Django's `EmailMessage` class.

Anyway, as someone not intimately familiar with how email messages are encoded, what's missing from the comment for me is the "why". The behavior of the if condition isn't complicated, so it's easy to look at it and read the code as "if there's a body or if the body_msg "is_multipart", then attach body_msg". Do you think it's obvious to someone more familiar with email stuff why these steps are needed? In that case, maybe a comment isn't needed.